### PR TITLE
adding optional IsReadyToPay field in entitlements

### PR DESCRIPTION
**Pull request by chenshay**

### CODE REVIEW

File: src/api/entitlements.js [hunk 1]
```diff
@@ -26,15 +26,21 @@ export class Entitlements {
...
+    service, raw, entitlements, currentProduct, ackHandler, opt_isReadyToPay) {
+
     /** @const {string} */
     this.service = service;
     /** @const {string} */
     this.raw = raw;
     /** @const {!Array<!Entitlement>} */
     this.entitlements = entitlements;
 
+    /** @const {boolean|undefined} */
```
Comment (dvoytenko) [inline src/api/entitlements.js]: Just `{boolean}`, since we will default it to `false` above.

File: src/api/entitlements.js [hunk 2]
```diff
@@ -26,15 +26,21 @@ export class Entitlements {
    * @param {!Array<!Entitlement>} entitlements
    * @param {?string} currentProduct
    * @param {function(!Entitlements)} ackHandler
+   * @param {boolean=} opt_isReadyToPay
```
Comment (dvoytenko) [inline src/api/entitlements.js]: Let's just name it `isReadyToPay`.

File: src/api/entitlements.js [hunk 3]
```diff
@@ -26,15 +26,21 @@ export class Entitlements {
...
    */
-  constructor(service, raw, entitlements, currentProduct, ackHandler) {
+  constructor(
```
Comment (dvoytenko) [inline src/api/entitlements.js]: Please do `isReadyToPay = false`

File: src/api/entitlements.js [hunk 4]
```diff
@@ -26,15 +26,21 @@ export class Entitlements {
...
+
     /** @const {string} */
     this.service = service;
     /** @const {string} */
     this.raw = raw;
     /** @const {!Array<!Entitlement>} */
     this.entitlements = entitlements;
 
+    /** @const {boolean|undefined} */
+    this.isReadyToPay = opt_isReadyToPay;
```
Comment (dvoytenko) [inline src/api/entitlements.js]: Please also pass it in the `clone()` and `json()` methods below.

File: src/api/entitlements.js (hunk 1 above)
Comment (chenshay) [reply src/api/entitlements.js]: done

File: src/api/entitlements.js (hunk 2 above)
Comment (chenshay) [reply src/api/entitlements.js]: done

File: src/api/entitlements.js (hunk 3 above)
Comment (chenshay) [reply src/api/entitlements.js]: done

File: src/api/entitlements.js (hunk 4 above)
Comment (chenshay) [reply src/api/entitlements.js]: done

**Pull request merged**